engine/src/test/java/org/hibernate/validator/test/cfg/ValidatorConstraintMappingTest.java

```diff
@@ -0,0 +1,308 @@
+		assertNumberOfViolations( violations, 1 );
+		assertCorrectValidatorTypes( violations, NonDefaultLongValidator.class );
+		
+		violations = validator.validate( new ConstrainedIntegerFieldBean() );
+		assertNumberOfViolations( violations, 1 );
+		assertCorrectValidatorTypes( violations, NonDefaultIntegerValidator.class );
+	}
+
+	@Test(
+			expectedExceptions = ValidationException.class,
+			expectedExceptionsMessageRegExp = "HV[0-9]*:"
+					+ " .*\\$ConstraintAnnotation is configured more than once via the programmatic constraint definition API."
+	)
+	public void testMultipleDefinitionForSameConstraintOnSameConstraintMapping() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.validatedBy( NonDefaultLongValidator.class )
+				.constraint( ConstraintAnnotation.class )
+				.includeExistingValidators( false )
+				.validatedBy( NonDefaultIntegerValidator.class );
+	}
+
+	@Test(
+			expectedExceptions = ValidationException.class,
+			expectedExceptionsMessageRegExp = "HV000167:.*"
+	)
+	public void testMultipleDefinitionForSameConstraintOnDifferentConstraintMappings() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.validatedBy( NonDefaultLongValidator.class );
+
+		ConstraintMapping otherMapping = config.createConstraintMapping();
+		otherMapping.constraint( ConstraintAnnotation.class )
+				.includeExistingValidators( false )
+				.validatedBy( NonDefaultIntegerValidator.class );
+		
+		config.addMapping( mapping ); 
+		config.addMapping( otherMapping );
+		config.buildValidatorFactory().getValidator();
+	}
+
+	@Test(
+			expectedExceptions = UnexpectedTypeException.class,
+			expectedExceptionsMessageRegExp = "HV000150:.*"
+	)
+	public void testMultipleValidatorsForSameType() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.includeExistingValidators( true )
+				.validatedBy( NonDefaultIntegerValidator.class );
+		
+		config.addMapping( mapping );
+		Validator validator = config.buildValidatorFactory().getValidator();
+		validator.validate( new ConstrainedIntegerFieldBean() );
+	}
+
+	@Target({ FIELD, METHOD, PARAMETER, ANNOTATION_TYPE })
+	@Retention(RUNTIME)
+	@Documented
+	private @interface NonConstraintAnnotation {
+	}
+
+	@Target({ FIELD, METHOD, PARAMETER, ANNOTATION_TYPE })
+	@Retention(RUNTIME)
+	@Documented
+	@Constraint(validatedBy = { DefaultIntegerValidator.class, DefaultStringValidator.class })
+	private @interface ConstraintAnnotation {
+		String message() default "Default ConstraintAnnotation violation message";
+
+		Class<?>[] groups() default { };
+
+		Class<? extends Payload>[] payload() default { };
+	}
+
+	/*
+	 * Implementation note: we are compiling in Java 6 mode, so {@code @SafeVarargs} is of no use here. Thus, we work
+	 * around the compiler warnings by not restraining the type of classes passed as parameters and just casting them.
+	 */
+	@SafeVarargs
+	@SuppressWarnings("unchecked")
+	private static void assertCorrectValidatorTypes(Set<? extends ConstraintViolation<?>> violations,
```

Comment:
Well, I could, but when someone reads the tests, that's not the meaning I want to convey: I just want to know which validators were executed, no matter their messages. Note that it's exactly how it's implemented, though.

Now, if it's bothering you, indeed I can just put a static final string in each validator type and go with `assertCorrectConstraintViolationMessages` inside each test.